tests/test_parser.py

```diff
@@ -419,6 +419,41 @@ def test_parser_request_4(self):
         with self.assertRaisesRegex(TypeError, 'a bytes-like object'):
             p.feed_data('POST  HTTP/1.2')
 
+    def test_parser_request_fragmented(self):
+        m = mock.Mock()
+        headers = {}
+        m.on_header.side_effect = headers.__setitem__
+        p = httptools.HttpRequestParser(m)
+
+        REQUEST = (
+            b'PUT / HTTP/1.1\r\nHost: localhost:1234\r\nContent-',
```

Comment:
I slightly changed the fragmented point and it now starts to be broken

Reply:
I think it was on purpose where the splitting was, especially the empty HTTP header.

Reply:
Another test which feeds byte-for-byte would also be nice.

Reply:
Thanks for the advice. I added a few more tests